jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshTemplate.java

```diff
@@ -51,59 +57,43 @@ public SshSetting getSshSetting() {
         notNull(startTime, "startTime cannot be null");
         notNull(endTime, "endTime cannot be null");
-        isTrue(!StringUtils.isEmpty(sshSetting.getLogPath()), "logPath must not be null or empty");
-        isTrue(!StringUtils.isEmpty(timestampFormat), "timestampFormat of log must not be null or empty");
+
         // if server has different time (ntp out of sync etc.)
-        Long serverEpochTime = Long.valueOf(executeCommand(Commands.EPOCH_TIME).trim());
-        Long timeOffset = serverEpochTime - ZonedDateTime.now().toEpochSecond();
+        Long serverEpochTime = valueOf(executeCommand(Commands.EPOCH_TIME).trim());
+        Long timeOffset = serverEpochTime - now().toEpochSecond();
 
         // if server has different time zone
         String timeZone = executeCommand(Commands.TIMEZONE).trim();
-        ZonedDateTime serverStartTime = startTime.withZoneSameInstant(ZoneId.of(timeZone)).plusSeconds(timeOffset);
-        ZonedDateTime serverEndTime = endTime.withZoneSameInstant(ZoneId.of(timeZone)).plusSeconds(timeOffset);
+        ZonedDateTime serverStartTime = startTime.withZoneSameInstant(of(timeZone)).plusSeconds(timeOffset);
+        ZonedDateTime serverEndTime = endTime.withZoneSameInstant(of(timeZone)).plusSeconds(timeOffset);
 
         List<String> logPaths = rollingLogResolver.resolveLogNames(sshSetting.getLogPath(), this, serverStartTime, serverEndTime);
         StringBuilder logs = new StringBuilder();
         for (String logPath : logPaths) {
             logs.append(copySingleLogFile(serverStartTime, serverEndTime, logPath));
         }
-
         return new SshLog(logs.toString(), sshSetting);
     }
 
     private String copySingleLogFile(final ZonedDateTime startTime, final ZonedDateTime endTime, final String logPath) throws IOException {
-        // get first and last line number
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(timestampFormat);
-        String firstLastLineCommand = String
-            .format(Commands.FIRST_LAST_LINES_BETWEEN_TIMESTAMPS, startTime.format(formatter), endTime.format(formatter), logPath);
-        String firstLastLines = executeCommand(firstLastLineCommand).trim();
-
-        String logContents = "";
-        if (!firstLastLines.isEmpty()) {
-            String[] lineNumbers = firstLastLines.split("\\s+");
-            Long startLine = Long.valueOf(lineNumbers[0]);
-            Long endLine = Long.valueOf(lineNumbers[1]);
-
-            // get lines between line numbers
-            String getLinesCommand = String.format(Commands.PRINT_BETWEEN_LINES, startLine, endLine, logPath);
-            logContents = executeCommand(getLinesCommand).trim();
-        }
-
-        return logContents;
+        DateTimeFormatter formatter = ofPattern(timestampFormat);
+        String getLogBetweenTimestamps = format(getLogBetweenTimestampsCommand,
+            startTime.format(formatter), endTime.format(formatter), endTime.plusYears(1).format(formatter), logPath);
```

Comment:
I would also add one minute and one second just to be safe in case that the user does not use timestamp with year and just uses minutes and seconds or just seconds like in unix timestamp.

```suggestion
            ZonedDateTime upperLimitTimestamp = endTime.plusYears(1).plusMinutes(1).plusSeconds(1);
            startTime.format(formatter), endTime.format(formatter), upperLimitTimestamp.format(formatter), logPath);
```